hoomd/pytest/test_dcd.py

```diff
@@ -0,0 +1,66 @@
+import hoomd
+import pytest
+import numpy as np
+
+
+def test_attach(simulation_factory, two_particle_snapshot_factory, tmp_path):
+    d = tmp_path / "sub"
+    d.mkdir()
```

Comment:
```suggestion
    d = tmp_path
```
`tmp_path` should already be a clean subdirectory - no need to create another subdirectory inside it.

Same for the other tests that follow pattern.